### ISSUE TITLE
Daily tip on SpaceMacs homescreen

### ISSUE THREAD

**Issue by geo7** [syl20bnr/spacemacs#1598]
In chat with tuhdo and TheBB subject of a daily tip came up. 

I leave gitter open a lot of the time as there is often a nice key command / function mentioned at some point that I can use. It might be nice to have something similar (non intrusively) on the homepage as well. This could (I presume) be linked to any relevant documentation.

**Comment by tuhdo** [comment]
@syl20bnr Prelude [implemented this already](https://github.com/bbatsov/prelude/blob/30fea6d9d6546e2cf055b91335758cdaad9038ac/core/prelude-core.el#L375), so we can use it as an example. Prelude shows the tip [5 seconds after init](https://github.com/bbatsov/prelude/blob/00b48933bd9107273943fa7e278d05c7dc6840b0/init.el#L131).

**Comment by geo7** [comment]
@tuhdo are their tips fixed or can the community add to them?

**Comment by tuhdo** [comment]
Prelude's tips are fixed. But they can add more useful ones with PR.

**Comment by syl20bnr** [comment]
This is an excellent idea. Like prelude we can take the `defvar` approach with a new layer variable called `layer-tips`. Then the engine should display randomly a tip for used layers only.
Tips can fit nicely with the current toggles we have on the homepage.

But for this section I see a necessary dotfile variable: `dotspacemacs-show-tips`. The idea is that the display of the tips is only driven by this variable so we don't have to remember the toggle state of the homepage button.

In all case a release notes that has not been viewed by the user has priority oven the display.

If I sum up the sections
- quickhelp:
  - shown by explicit usage of the toggle button
  - not persistent accros session
- tips:
  - shown by looking at the `dotspacemacs-show-tips` variable
  - can be toggled with the button - not persistent accross session (must use the dotfile variable)
- release notes:
  - driven by the current spacemacs version number and stored value in `.cache` folder
  - can be toggled with the button - persistent toggle state accross session
   - when it should be displayed it takes precedence over any other sections.

**Comment by syl20bnr** [comment]
If we limit the tips to only short sentences then it could be a section independent from quickhelp and release-notes. We could display the tip on top of these action just beneath the button bar.

**Comment by srid** [comment]
@BrianHicks created a tips site: http://spacemacs.brianthicks.com/

I like the mini animated gif demonstration. It would be nice if `<layer>-tips` contained them, to be rendered inside emacs as animated gifs.

**Comment by BrianHicks** [comment]
I'm happy to record any that the community wants... probably using the official spacemacs theme, eh? ;)

**Comment by CestDiego** [comment]
it would be awesome if we get the tips in a short format in the spacemacs home buffer, and with a button that says: read whole tip which takes the user to the long description of the tip and the animation (if there is one available)

**Comment by CestDiego** [comment]
also consider opening new frames with a random tip instead of scratch? < make this configurable

**Comment by a13ph** [comment]
Sacha has a lot on similar topic - using org-drill for spaced repetition of emacs commands and other stuff.

https://punchagan.muse-amuse.in/posts/org-drill-for-making-it-stick.html (not sacha, but same thing)

http://sachachua.com/blog/2013/02/rediscovering-emacs-features-or-what-to-do-after-you-get-carried-away-installing-packages/

http://sachachua.com/blog/2005/01/more-emacs-evangelization-flashcard/

http://sachachua.com/blog/2016/02/building-today-learned-habit-displaying-documentation-random-emacs-commands/

tldr: make sure people will *understand* well contents of tips... somehow

**Comment by rommeswi** [comment]
I just came across this as I was googling for this functionality.... It would be really great for this to be implemented!

**Comment by rommeswi** [comment]
I still think this would be an incredibly useful feature. Perhaps it can be implemented in the same way as the banners?

Imo the best option would be if the tip would be based on the major mode of a random file in recentf. It can be as simple as displaying a binding. For example, if a `.py` file is randomly selected from recentf, the daily tip might randomly draw `, r b`, display the binding and the command that is run. The command has a link that opens the relevant help file.

The advantage of this approach is that it does not require the maintainer to curate a list of daily tips.

**Comment by smile13241324** [comment]
This sounds useful still I think we have more important tickets to work on right now. If someone would like to invest some time I would be open for a PR, just saying ;)
